otelcolbuilder/cmd/configprovider.go

```diff
@@ -45,9 +39,8 @@ func UseCustomConfigProvider(params *otelcol.CollectorSettings) error {
 	if err != nil {
 		return fmt.Errorf("setting feature gate flags failed: %s", err)
 	}
-	// to create the provider, we need config locations passed in via the command line
-	// to get these, we take the command the service uses to start, parse the flags, and read the values
-	flagset := flags(featuregate.GlobalRegistry())
+	// we need to check if any config locations have been set alongside the opamp config
+	flagset := flags()
```

Comment:
is there a reason that we don't need to pass in the featuregate anymore?

Reply:
We never did on Linux and Mac, because they're parsed again later anyway. But it looks like we still may need to on Windows.

Reply:
Don't need to do it on Windows either. We only need to include flags here whose input we need, which means just the config ones.